Observer/Config.php

```diff
@@ -10,27 +10,29 @@ class Config implements ObserverInterface
     public function __construct(
         \Magento\Framework\Message\ManagerInterface $messageManager,
         \Magento\Framework\App\Request\Http         $request,
-        \Metrilo\Analytics\Helper\Data              $helper,
+        \Metrilo\Analytics\Helper\Data              $dataHelper,
+        \Metrilo\Analytics\Helper\Activity          $activityHelper,
         \Metrilo\Analytics\Helper\ApiClient         $apiClient
     ) {
         $this->messageManager = $messageManager;
         $this->request        = $request;
-        $this->helper         = $helper;
+        $this->dataHelper     = $dataHelper;
+        $this->activityHelper = $activityHelper;
         $this->apiClient      = $apiClient;
     }
 
     public function execute(Observer $observer)
     {
         try {
             $storeId  = (int)$this->request->getParam('store', 0);
-            $activity = $this->helper->createActivity($storeId, 'integrated');
+            $activity = $this->activityHelper->createActivity($storeId, 'integrated', $this->dataHelper->getApiToken($storeId), $this->dataHelper->getApiSecret($storeId));
```

Comment:
what is the point in passing these from the outside instead of getting them in the `ActivityHelper`?

Reply:
Currently all module logic for config getters lies in Data helper and same helper is used in the observer for the logger method. To acquire that data in the Activity helper I will have to add the Data helper in Activity constructor or create new methods inside the Activity class to get these config values from DB.

Reply:
> To acquire that data in the Activity helper I will have to add the Data helper in Activity constructor
is this a problem?

Reply:
I tough it's not good to have data helper in config observer constructor and in the same time have it in activity constructor where activity is part of config observer constructor aswell, but i just tested it and there are no errors so i guess it's okay.